src/controllers/movies.js

```diff
@@ -0,0 +1,114 @@
+const { Prisma } = require("@prisma/client");
+const prisma = require("../utils/prisma");
+
+const { MissingFieldsError } = require("../utils/errors");
+const { movie } = require("../utils/prisma");
+
+const getMovies = async (req, res) => {
+  const { runtimeLt, runtimeGt } = req.query;
+
+  const movies = await prisma.movie.findMany({
+    where: {
+      runtimeMins: {
+        lt: runtimeLt ? Number(runtimeLt) : undefined,
+        gt: runtimeGt ? Number(runtimeGt) : undefined,
+      },
+    },
+    include: {
+      screenings: true,
+    },
+  });
+  res.json({ movies: movies });
+};
+
+const createMovie = async (req, res) => {
+  const { title, runtimeMins, screenings } = req.body;
+  let includeScreenings;
+  if (!title || !runtimeMins) {
+    throw new MissingFieldsError();
+  }
+
+  if (screenings) {
+    includeScreenings = {
+      createMany: {
+        data: screenings.map(({ screenId, startsAt }) => ({
+          screenId,
+          startsAt: new Date(startsAt),
+        })),
+      },
+    };
+  }
+
+  const createdMovie = await prisma.movie.create({
+    data: {
+      title: title,
+      runtimeMins: runtimeMins,
+      screenings: screenings ? includeScreenings : undefined,
+    },
+    include: { screenings: true },
+  });
```

Comment:
This is good, a small improvement here would be to remove the `let includeScreenings` and attach the createMany to the data object in the `if` statement, like this:

```js
  if (!title || !runtimeMins) {
    throw new MissingFieldsError();
  }

 const data = { titletitle, runtimeMins }

  if (screenings) {
    data.screenings = {
      createMany: {
        data: screenings.map(({ screenId, startsAt }) => ({
          screenId,
          startsAt: new Date(startsAt),
        })),
      },
    };
  }

  const createdMovie = await prisma.movie.create({
    data,
    include: { screenings: true },
  });
```